### PR TITLE
option object shouldn't be updated internally.

### PR DESCRIPTION
**Pull request by smihica**
I got "not-null constraint" error because the previous save(opt) call inserted "fields" into opt.
This may cause confusing errors.

### CODE REVIEW

**Comment by smihica** [comment]
This pr is for the following bug.
https://github.com/sequelize/sequelize/issues/4011

File: lib/instance.js [hunk 1]
```diff
@@ -477,7 +477,7 @@ Instance.prototype.save = function(options) {
   if (arguments.length > 1) {
     throw new Error('The second argument was removed in favor of the options object.');
   }
-  options = _.defaults(options || {}, {
+  options = _.defaults({}, options || {}, {
```
Comment (janmeier) [inline lib/instance.js]: I don't think we need  `options || {}`, `options` should be fine

File: lib/instance.js [hunk 2]
```diff
@@ -818,7 +818,7 @@ Instance.prototype.destroy = function(options) {
       for(var i = 0; i < primaryKeys.length; i++) {
           where[this.Model.rawAttributes[primaryKeys[i]].field] = this.get(primaryKeys[i], { raw: true });
       }
-      return this.sequelize.getQueryInterface().delete(this, this.Model.getTableName(options), where, _.defaults(options, { type: QueryTypes.DELETE,limit: null}));
+      return this.sequelize.getQueryInterface().delete(this, this.Model.getTableName(options), where, _.defaults({}, options, { type: QueryTypes.DELETE,limit: null}));
```
Comment (janmeier) [inline lib/instance.js]: Can't we just use `options` here?

**Comment by janmeier** [comment]
Looks good @smihica - do you mind adding a few tests. Preferrably unit tests, since they are faster. Something along the lines of 

```js
stub(sequelize, 'query').returns(Promise.resolve([]));

var options = { ... }

model.reload(options);

expect(options).to.deep.equal({... }
```


If you have too many problems trying to stub sequelize.query, integration tests that hit the db will be fine as well

File: lib/instance.js (hunk 2 above)
Comment (smihica) [inline lib/instance.js]: yeah I think so. thank you.

File: lib/instance.js (hunk 1 above)
Comment (smihica) [inline lib/instance.js]: Oh, exactly.

**Comment by smihica** [comment]
Yes, I'll try to add some tests later. thanks reviewing.

**Comment by smihica** [comment]
@janmeier 
OK. I added tests and rebased. and All unit and integration tests were passed in my environment.

**Comment by janmeier** [comment]
Solid work on the tests @smihica ! Thanks for this, and the other PR also :)

**Pull request merged**